mode/clojure/clojure.js

```diff
@@ -137,23 +137,18 @@ CodeMirror.defineMode("clojure", function (options) {
       "with-local-vars", "with-meta", "with-open", "with-out-str",
       "with-precision", "with-redefs", "with-redefs-fn", "xml-seq", "zero?",
       "zipmap"];
-  var commonIndentSymbols = [
-      "assoc", "binding", "bound-fn", "case", "catch", "comment", "cond",
-      "condp", "def", "defmethod", "defn", "defprotocol", "defrecord",
-      "defstruct", "deftype", "do", "doseq", "dotimes", "doto", "extend",
-      "extend-protocol", "extend-type", "fn", "for", "future", "if", "if-let",
-      "if-not", "let", "letfn", "locking", "loop", "ns", "proxy", "reify",
-      "struct-map", "try", "when", "when-first", "when-let", "when-not",
-      "when-some", "while", "with-open", "with-precision"];
+  var commonIndentCoreSymbols = [
+      "assoc", "binding", "bound-fn", "case", "catch", "comment", "fn", "for",
+      "future", "locking", "loop", "ns", "proxy", "reify", "struct-map", "try"];
 
   CodeMirror.registerHelper("hintWords", "clojure",
       commonAtoms.concat(commonSpecialForms, commonCoreSymbols));
 
   var atom = createLookupMap(commonAtoms);
   var specialForm = createLookupMap(commonSpecialForms);
   var coreSymbol = createLookupMap(commonCoreSymbols);
-  var indentSymbol = createLookupMap(commonIndentSymbols);
-  var assumeBody = /^(?:def|with)[^\/]+$|\/(?:def|with)/;
+  var indentCoreSymbols = createLookupMap(commonIndentCoreSymbols);
+  var assumeBody = /^cond|^def|^do|^extend|^if|^let|^when|^while|^with|\/(?:def|with)/;
```

Comment:
I think it'd make sense to use a `\b` marker after/around these words, so that things like `letter-at` or `default-for` (just making things up) don't match

Reply:
Good point! I’ll take care of that later today. Thanks!

Reply:
Welp, it turned out this is not as easy as I originally thought. I need to spend more time to think about how to do this right. I'll close this PR for now. Thanks!